opentracing-api-extensions-tracer/src/main/java/io/opentracing/contrib/api/tracer/APIExtensionsSpanBuilder.java

```diff
@@ -0,0 +1,132 @@
+
+public class APIExtensionsSpanBuilder implements SpanBuilder {
+
+    private final Tracer tracer;
+    private final List<TracerObserver> observers;
+
+    private final String operationName;
+    private final SpanBuilder wrappedBuilder;
+    private long startTimestampMicro = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis());
+    private long startTimeNano = System.nanoTime();
+    private final Map<String,Object> tags = new ConcurrentHashMap<String,Object>();
+
+    public APIExtensionsSpanBuilder(Tracer tracer, List<TracerObserver> observers,
+            String operationName, SpanBuilder builder) {
+        this.tracer = tracer;
+        this.observers = observers;
+        this.operationName = operationName;
+        this.wrappedBuilder = builder;
+    }
+
+    @Override
+    public SpanBuilder asChildOf(SpanContext parent) {
+        wrappedBuilder.asChildOf(parent);
+        return this;
+    }
+
+    @Override
+    public SpanBuilder asChildOf(BaseSpan<?> parent) {
+        wrappedBuilder.asChildOf(parent);
+        return this;
+    }
+
+    @Override
+    public SpanBuilder addReference(String referenceType, SpanContext referencedContext) {
+        wrappedBuilder.addReference(referenceType, referencedContext);
+        return this;
+    }
+
+    @Override
+    public SpanBuilder ignoreActiveSpan() {
+        wrappedBuilder.ignoreActiveSpan();
+        return this;
+    }
+
+    @Override
+    public SpanBuilder withTag(String key, String value) {
+        tags.put(key, value);
+        wrappedBuilder.withTag(key, value);
+        return this;
+    }
+
+    @Override
+    public SpanBuilder withTag(String key, boolean value) {
+        tags.put(key, value);
+        wrappedBuilder.withTag(key, value);
+        return this;
+    }
+
+    @Override
+    public SpanBuilder withTag(String key, Number value) {
+        tags.put(key, value);
+        wrappedBuilder.withTag(key, value);
+        return this;
+    }
+
+    @Override
+    public SpanBuilder withStartTimestamp(long microseconds) {
+        wrappedBuilder.withStartTimestamp(microseconds);
+        // Reset the nano start time, so that duration will be calculated based on explicitly
+        // provided timestamps
+        this.startTimeNano = 0;
+        return this;
+    }
+
+    @Override
+    public ActiveSpan startActive() {
+        return tracer.makeActive(startManual());
+    }
+
+    @Override
+    public Span startManual() {
+        APIExtensionsSpan span = new APIExtensionsSpan(wrappedBuilder.startManual(),
+                operationName, startTimestampMicro, startTimeNano, tags);
+        if (!observers.isEmpty()) {
+            for (TracerObserver observer : observers) {
+                span.addSpanObserver(observer.onStart(span));
+            }
+        }
+        return span;
+    }
+
+    @Override
+    public Span start() {
+        return startManual();
+    }
+
+    Map<String, Object> tags() {
+        return tags;
```

Comment:
@objectiser Can't you fine another way to test, _without exposing internals_ ?
I think the question is justified why this method exists.